samples/UserPassOrTokenClient.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+"""
+Get a users room list and indicate login type.
+
+to use user+pass to login and get a token:
+    arg usage: --host 'host' --user 'username' --password 'password'
+to use user+token to login:
+    arg usage: --host 'host' --user 'username' --token 'token'
+"""
+
+import argparse
+from matrix_client.client import MatrixClient
+
+
+def _example(host, user, password, token):
+    """run the example."""
+    client = None
+    if token:
+        print('token login')
+        client = MatrixClient(host, token=token, user_id=user)
+    else:
+        print('password login')
+        client = MatrixClient(host)
+        token = client.login_with_password(user, password)
+        print('got token: %s' % token)
+    for room_id, room in client.get_rooms().items():
+        print("is in room %s" % room_id)
+
+
+def main():
+    """Main entry."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--host", type=str, required=True)
+    parser.add_argument("--user", type=str, required=True)
+    parser.add_argument("--password", type=str)
+    parser.add_argument("--token", type=str)
+    args = parser.parse_args()
+    if not args.password and not args.token:
+        print('password or token is required')
+        exit(1)
```

Comment:
this is only not "outdated" because I've commented at the top of the file and that actual segment remains unchanged though there is now a comment and specific problem/exception handling for others.